src/app/machines/views/MachineDetails/MachineNetwork/NetworkTable/NetworkTableActions/NetworkTableActions.test.tsx

```diff
@@ -42,389 +45,311 @@ describe("NetworkTableActions", () => {
-            n.text() === "Mark as connected"
-        )
-        .exists()
-    ).toBe(true);
+      screen.getByRole("button", { name: "Mark as connected" })
+    ).toBeInTheDocument();
   });
 
-  it("can display an item to mark an interface as disconnected", () => {
+  it("can display an item to mark an interface as disconnected", async () => {
     nic.type = NetworkInterfaceTypes.PHYSICAL;
     nic.link_connected = true;
     const store = mockStore(state);
-    const wrapper = mount(
-      <Provider store={store}>
-        <NetworkTableActions
-          nic={nic}
-          setExpanded={jest.fn()}
-          systemId="abc123"
-        />
-      </Provider>
+    renderWithMockStore(
+      <NetworkTableActions
+        nic={nic}
+        setExpanded={jest.fn()}
+        systemId="abc123"
+      />,
+      { store }
     );
     // Open the menu:
-    wrapper.find("Button.p-contextual-menu__toggle").simulate("click");
-    wrapper.update();
+    await openMenu();
+
     expect(
-      wrapper
-        .findWhere(
-          (n) =>
-            n.type() === "button" &&
-            n.hasClass("p-contextual-menu__link") &&
-            n.text() === "Mark as disconnected"
-        )
-        .exists()
-    ).toBe(true);
+      screen.getByRole("button", { name: "Mark as disconnected" })
+    ).toBeInTheDocument();
   });
 
-  it("does not display an item to mark an alias as connected", () => {
+  it("does not display an item to mark an alias as connected", async () => {
     nic.type = NetworkInterfaceTypes.PHYSICAL;
     nic.link_connected = false;
     const link = networkLinkFactory();
     nic.links = [networkLinkFactory(), link];
     const store = mockStore(state);
-    const wrapper = mount(
-      <Provider store={store}>
-        <NetworkTableActions
-          link={link}
-          nic={nic}
-          setExpanded={jest.fn()}
-          systemId="abc123"
-        />
-      </Provider>
+    renderWithMockStore(
+      <NetworkTableActions
+        link={link}
+        nic={nic}
+        setExpanded={jest.fn()}
+        systemId="abc123"
+      />,
+      { store }
     );
     // Open the menu:
-    wrapper.find("Button.p-contextual-menu__toggle").simulate("click");
-    wrapper.update();
+    await openMenu();
     expect(
-      wrapper
-        .findWhere(
-          (n) =>
-            n.type() === "button" &&
-            n.hasClass("p-contextual-menu__link") &&
-            n.text() === "Mark as connected"
-        )
-        .exists()
-    ).toBe(false);
+      screen.queryByRole("button", { name: "Mark as connected" })
+    ).not.toBeInTheDocument();
   });
 
-  it("does not display an item to mark an alias as disconnected", () => {
+  it("does not display an item to mark an alias as disconnected", async () => {
     nic.type = NetworkInterfaceTypes.PHYSICAL;
     nic.link_connected = true;
     const link = networkLinkFactory();
     nic.links = [networkLinkFactory(), link];
     const store = mockStore(state);
```

Comment:
There are no assertions run on the store and this can be removed. You can pass the state directly to `renderWithMockStore`